## Load the standard libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

## Load the data

In [2]:
data = pd.read_csv('Health_Insurance.csv')
data.head()

Age  Purchased
0   19          0
1   35          0
2   26          0
3   27          0
4   19          0

## Seperate X and y

In [3]:
X = data.drop('Purchased', axis = 1)
y = data['Purchased']

#### We First will create 1 train and 1 test set using train_test_split() to understand which of Logistic Regression or KNeighborsClassifier works better on the data. 

#### Once we get a best algorithm then we create multiple train and test sets to understand which train and test set will give the highest accuracy

In [8]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

## Apply Logistic Regression and check accuracy

In [5]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression()
lr

LogisticRegression()

In [9]:
lr.fit(X_train, y_train)

LogisticRegression()

In [10]:
y_pred_lr = lr.predict(X_test)

In [11]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test, y_pred_lr)

0.9

#### Note: Logistic Regression Algorithm is producing an accuracy of 90%

In [12]:
from sklearn.neighbors import KNeighborsClassifier
knc = KNeighborsClassifier()
knc

KNeighborsClassifier()

In [13]:
knc.fit(X_train, y_train)

KNeighborsClassifier()

In [14]:
y_pred_knc = knc.predict(X_test)

In [15]:
accuracy_score(y_test, y_pred_knc)

0.85

#### KNeighbors Classifier is producing an accuracy of 85%.

## Looking at the accuracies we conclude that Logistic Regression is performing better on the data

In [16]:
from sklearn.model_selection import cross_val_score
cross_val_score(lr, X, y, cv = 10)

array([0.95 , 0.65 , 0.825, 0.925, 0.925, 0.825, 0.75 , 0.75 , 0.8  ,
       0.975])

## the highest accuracy on the health insurance data is 97.5% using LogisticRegression and that too on the 10th train and test set